test/test_encryption.py

```diff
@@ -2687,5 +2687,180 @@ def test_int(self):
         self.run_test_cases("Int", RangeOpts(min=0, max=200, sparsity=1), int)
 
 
+# https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/README.rst#automatic-data-encryption-keys
+class TestAutomaticDecryptionKeys(EncryptionIntegrationTest):
+    @client_context.require_no_standalone
+    @client_context.require_version_min(6, 0, -1)
+    def setUp(self):
+        super().setUp()
+        self.key1_document = json_data("etc", "data", "keys", "key1-document.json")
+        self.key1_id = self.key1_document["_id"]
+        self.client.drop_database(self.db)
+        self.key_vault = create_key_vault(self.client.keyvault.datakeys, self.key1_document)
+        self.addCleanup(self.key_vault.drop)
+        self.key_vault_client = self.client
+        self.client_encryption = ClientEncryption(
+            {"local": {"key": LOCAL_MASTER_KEY}},
+            self.key_vault.full_name,
+            self.key_vault_client,
+            OPTS,
+        )
+        self.addCleanup(self.client_encryption.close)
+        self.client = rs_or_single_client()
+        self.client.drop_database("db")
+        self.db = self.client.db
+        self.addCleanup(self.client.close)
```

Comment:
I think we can remove these last 4 lines and reuse the existing self.client (ClientContext.client).

Reply:
Done.